tools/autograd/load_derivatives.py

```diff
@@ -516,6 +517,15 @@ def repl(m: Match[str]) -> str:
 
             formula = re.sub(regex.format(name), repl, formula)
 
+        # c10::optional<std::string> types stored in Backward nodes must be
+        # converted to c10::optional<c10::string_view> before being passed into
+        # the backward function
+        if nctype.type == OptionalCType(BaseCType(stringT)):
```

Comment:
I'm not 100% sure if this is the best way to do this or not. The above replacements don't consider the type when deciding whether to replace or not. So I don't think this replacement should happen there, unless we add an optional type detection above.

A different solution, which wouldn't mess with codegen, might be to add a constructor for `optional<c10::string_view>` which can take `optional<std::string>` as its argument. I'm not sure if that would be better or not

Reply:
This seems fine. Adding a constructor for optional is not good, std::optional wouldn't let you do that